src/t_zset.c

```diff
@@ -2826,8 +2829,10 @@ void zunionInterDiffGenericCommand(client *c, robj *dstkey, int numkeysIndex, in
                 server.dirty++;
             }
         }
+        decrRefCount(dstobj);
     } else if (cardinality_only) {
         addReplyLongLong(c, cardinality);
+        decrRefCount(dstobj);
```

Comment:
maybe that one should be lazy freed as well?

Reply:
actually cardinality_only (ZINTERCARD) does not touch dstobj at all. so the obj is a small obj. (we can even not create it in this case)

Reply:
and it seems that some performance can be improved (in the simplest scenario)

src/redis-cli zadd zset 1 a
src/redis-benchmark -P 100 -n 4000000 zintercard 1 zset

unstable:
```
Summary:
  throughput summary: 784160.00 requests per second
  latency summary (msec):
          avg       min       p50       p95       p99       max
        6.363     4.072     6.183     7.031    11.015    13.687
```

remove the useless create:
```
Summary:
  throughput summary: 863371.50 requests per second
  latency summary (msec):
          avg       min       p50       p95       p99       max
        5.776     2.792     5.543     7.263    10.359    12.815
```

Reply:
ohh, right. i was thinking that we do maintain some collection just for the carnality computation, but we don't.

regarding the 10% savings of not allocating it, is it an extreme case because the set is so small?

Reply:
> regarding the 10% savings of not allocating it, is it an extreme case because the set is so small?

It also applies to other situations, like this one is a 120 elements zset: 204708.30 -> 228571.42

Reply:
yeah, let's do it.

Reply:
Actually for my example of setnum == 1, we can do a fast return (this also can work for SINTERCARD):
```
if (op == SET_OP_INTER && cardinality_only && setnum == 1) {
        addReplyLongLong(c, zuiLength(&src[0]));
        zfree(src);
        return;
    }
```

Reply:
btw, i see you updated the top comment, i updated the top comment at the same time as you...